src/transformers/generation/utils.py

```diff
@@ -342,6 +343,107 @@ class GenerateBeamEncoderDecoderOutput(ModelOutput):
 GenerateOutput = Union[GenerateNonBeamOutput, GenerateBeamOutput]
 
 
+def _cache_dependant_input_preparation(
```

Comment:
Request: let's move this two inside `GenerationMixin`. Then, all models that overwrite `prepare_inputs_for_generation` can access these functions without additional imports. (See global comment)

Reply:
done